src/entities/fractions/fraction.rs

```diff
@@ -82,19 +91,20 @@ where
     }
 
     // Converts the fraction to a string with a specified number of significant digits and rounding strategy
-    fn to_significant(&self, significant_digits: u8, rounding: Rounding) -> String {
-        assert!(
-            significant_digits > 0,
-            "Significant digits must be positive."
-        );
-
+    fn to_significant(&self, significant_digits: u8, rounding: Rounding) -> Result<String, Error> {
+        if significant_digits == 0 {
+            return Err(Error::Incorrect(format!(
+                "significant digits:{} should always be positive",
+                significant_digits
```

Comment:
The error message will be "significant digits:0 should always be positive". It's not informative.

Reply:
I mean the dynamic variable in the formatted string doesn't make sense. The text should be "significant digits cannot be 0".